python/feathub/feature_views/transforms/python_udf_transform.py

```diff
@@ -24,14 +24,48 @@ class PythonUdfTransform(Transformation):
     time.
     """
 
-    def __init__(self, udf: Callable[[pd.Series], Any]) -> None:
+    def __init__(
+        self,
+        udf: Callable[[pd.Series], Any],
+        fail_on_exception: bool = True,
+        default_value: Any = None,
+    ) -> None:
         """
         :param udf: The udf that will be invoked for each row. The input
                           of the udf is a Pandas Series object that represent the
                           row.
+        :param fail_on_exception: Whether to fail the job in case of an exception is
+                                  raised by the udf.
+        :param default_value: If `fail_on_exception` is set to false, this is the
+                              default value of the result when an exception is raised.
```

Comment:
Would it be better if Feathub checks whether the default value meets its data type in schema? For example, Feathub may throw exception in the following situations:
- data type is boolean, but the default value is None
- data type is integer, but the default value is "foobar".

Reply:
Currently, if the default value is incompatible with the data type of the feature, the exception will be thrown at runtime by the processor. It is a good improvement to validate in Feathub and fail earlier. The improvement is not trivial so I added a TODO. What do you think?